### PR TITLE
Add index generator

### PR DESCRIPTION
**Pull request by srchase**
Generate index.ts with exports from Service and Commands.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/IndexGenerator.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import software.amazon.smithy.model.shapes.ServiceShape;
+
+/**
+ * Generates an index to export the service client and each command.
+ */
+final class IndexGenerator {
+
+    private IndexGenerator() {}
+
+    static void writeIndex(
```
Comment (kstich) [inline smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/IndexGenerator.java]: Can you add a couple comments to the body here to explain what's being written?

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/IndexGenerator.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        TypeScriptSettings settings,
+        Model model,
+        SymbolProvider symbolProvider,
+        FileManifest fileManifest
+    ) {
+        TypeScriptWriter writer = new TypeScriptWriter("");
+        ServiceShape service = settings.getService(model);
+        Symbol symbol = symbolProvider.toSymbol(service);
+        writer.write("export * from \"./" + symbol.getName() + "\";");
```
Comment (kstich) [inline smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/IndexGenerator.java]: Does this need to export the [non-modular client](https://github.com/awslabs/smithy-typescript/blob/master/smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CodegenVisitor.java#L232)?

Comment (srchase) [reply smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/IndexGenerator.java]: Added in https://github.com/awslabs/smithy-typescript/pull/34/commits/6046592d5d10cc36f6420ba3faab29bed072578a

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/IndexGenerator.java (hunk 1 above)
Comment (srchase) [reply smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/IndexGenerator.java]: added a couple commends in https://github.com/awslabs/smithy-typescript/pull/34/commits/6046592d5d10cc36f6420ba3faab29bed072578a

**Pull request merged**